utils/src/blocknative.rs

```diff
@@ -0,0 +1,150 @@
+//! Copy from <https://github.com/gakonst/ethers-rs/blob/master/ethers-middleware/src/gas_oracle/blocknative.rs>
+//! which is unmaintained and out-of-sync with the latest blocknative feed
+//!
+//! TDOO: revisit this or remove this when switching to `alloy-rs`
+use async_trait::async_trait;
+use ethers::{
+    middleware::gas_oracle::{from_gwei_f64, GasCategory, GasOracle, GasOracleError, Result},
+    types::U256,
+};
+use reqwest::{header::AUTHORIZATION, Client};
+use serde::Deserialize;
+use std::collections::HashMap;
+use url::Url;
+
+const URL: &str = "https://api.blocknative.com/gasprices/blockprices";
```

Comment:
checking out rate limits on [blocknative](https://docs.blocknative.com/gas-prediction/gas-platform#rate-limits), they say the following
```
Each endpoint is limited to provide updated results once per second for paid tier API keys, or every 5 seconds for free tier API keys. Polling faster than these rates may deliver stale data.
``` 
Are we on the free tier and i reckon we're not likely to hit the rate limit?

Reply:
the rate limit is not great, but our prover is only sending once per hour, we are not hitting this limit if we are just querying it once and send tx with it. But if later we decided to send multiple queries in sequence to wait for favorable gas, then that logic might require paid tier. 

honestly, right now, i just want to merge something simple that at least provide some lower gas price setting. We can also constantly monitor the datadog log, if we see too many rate limited errors, then we can consider upgrading from free tier.

Reply:
got it! the rate limit for the free tier sounds sufficient for now